cannot build on windows10

accoding to Specific for Windows, tried to build with node-gyp, it failed:

gyp info it worked if it ends with ok
gyp info using node-gyp@3.2.0
gyp info using node@4.2.2 | win32 | x64
gyp info spawn C:\Python27\python.EXE
gyp info spawn args [ 'C:\\Users\\zrs\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=2013',
gyp info spawn args   '-I',
gyp info spawn args   'D:\\dev\\opensource\\node-opencv\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\zrs\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\zrs\\.node-gyp\\4.2.2\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\zrs\\.node-gyp\\4.2.2',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Users\\zrs\\AppData\\Roaming\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=D:\\dev\\opensource\\node-opencv',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'D:\\dev\\opensource\\node-opencv\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn msbuild
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64' ]
在此解决方案中一次生成一个项目。若要启用并行生成，请添加“/m”开关。
  init.cc
  Matrix.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\init.cc) [D:\dev\op
ensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\init.cc) [D:\dev\op
ensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\Matrix.cc) [D:\dev\
opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\init.cc) [D:\dev\o
pensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\Matrix.cc) [D:\dev\
opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\Matrix.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
..\src\Matrix.cc(317): warning C4267: “初始化”: 从“size_t”转换到“int”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv.vcxpr
oj]
..\src\Matrix.cc(653): warning C4267: “参数”: 从“size_t”转换到“uint32_t”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv.v
cxproj]
..\src\Matrix.cc(691): warning C4267: “参数”: 从“size_t”转换到“uint32_t”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv.v
cxproj]
..\src\Matrix.cc(1224): warning C4267: “参数”: 从“size_t”转换到“uint32_t”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv.
vcxproj]
..\src\Matrix.cc(1997): warning C4267: “=”: 从“size_t”转换到“unsigned int”，可能丢失数据 [D:\dev\opensource\node-opencv\build\open
cv.vcxproj]
  OpenCV.cc
  CascadeClassifierWrap.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\OpenCV.cc) [D:\dev\
opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\OpenCV.cc) [D:\dev\
opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\CascadeClassifierWr
ap.cc) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\CascadeClassifierWr
ap.cc) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\OpenCV.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\CascadeClassifierW
rap.cc) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
..\src\OpenCV.cc(44): warning C4267: “初始化”: 从“size_t”转换到“unsigned int”，可能丢失数据 [D:\dev\opensource\node-opencv\build\open
cv.vcxproj]
  Contours.cc
  Point.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\Contours.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\Contours.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\Contours.cc) [D:\d
ev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\Point.cc) [D:\dev\o
pensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\Point.cc) [D:\dev\o
pensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\Point.cc) [D:\dev\
opensource\node-opencv\build\opencv.vcxproj]
  VideoCaptureWrap.cc
..\src\Contours.cc(86): warning C4267: “参数”: 从“size_t”转换到“uint32_t”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv.
vcxproj]
..\src\Contours.cc(306): warning C4267: “参数”: 从“size_t”转换到“uint32_t”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv
.vcxproj]
..\src\Contours.cc(308): warning C4267: “参数”: 从“size_t”转换到“uint32_t”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv
.vcxproj]
..\src\Contours.cc(319): warning C4267: “参数”: 从“size_t”转换到“uint32_t”，可能丢失数据 [D:\dev\opensource\node-opencv\build\opencv
.vcxproj]
  CamShift.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\VideoCaptureWrap.cc
) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\VideoCaptureWrap.cc
) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\VideoCaptureWrap.c
c) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\CamShift.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\CamShift.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\CamShift.cc) [D:\d
ev\opensource\node-opencv\build\opencv.vcxproj]
  HighGUI.cc
  FaceRecognizer.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\HighGUI.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\HighGUI.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\HighGUI.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\FaceRecognizer.cc)
[D:\dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\FaceRecognizer.cc)
[D:\dev\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\FaceRecognizer.cc)
 [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
  Features2d.cc
  BackgroundSubtractor.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\Features2d.cc) [D:\
dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\Features2d.cc) [D:\
dev\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\Features2d.cc) [D:
\dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\BackgroundSubtracto
r.cc) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\BackgroundSubtracto
r.cc) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\BackgroundSubtract
or.cc) [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
  Constants.cc
..\src\BackgroundSubtractor.cc(85): warning C4267: “初始化”: 从“size_t”转换到“unsigned int”，可能丢失数据 [D:\dev\opensource\node-ope
ncv\build\opencv.vcxproj]
  Calib3D.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\Constants.cc) [D:\d
ev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\Constants.cc) [D:\d
ev\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\Constants.cc) [D:\
dev\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\Calib3D.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\Calib3D.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\Calib3D.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
..\src\Constants.cc(5): warning C4005: “CONST”: 宏重定义 [D:\dev\opensource\node-opencv\build\opencv.vcxproj]
          C:\Program Files (x86)\Windows Kits\8.1\Include\shared\minwindef.h(153) : 参见“CONST”的前一个定义
  ImgProc.cc
  Stereo.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\ImgProc.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\ImgProc.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\ImgProc.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\Stereo.cc) [D:\dev\
opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\Stereo.cc) [D:\dev\
opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\Stereo.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
  LDAWrap.cc
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(48): error C2039: “type_info”: 不是“std”的成员 (..\src\LDAWrap.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(60): error C2039: “type_info”: 不是“std”的成员 (..\src\LDAWrap.cc) [D:\dev
\opensource\node-opencv\build\opencv.vcxproj]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(80): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::typed_base_any_poli
  cy<T>”的引用
          with
          [
              T=cvflann::flann_centers_init_t
          ]
          e:\opencv\2.4.9\build\include\opencv2\flann\any.h(102): 参见对正在编译的类 模板 实例化“cvflann::anyimpl::big_any_policy<cvf
  lann::flann_centers_init_t>”的引用
e:\opencv\2.4.9\build\include\opencv2\flann\any.h(288): error C2039: “type_info”: 不是“std”的成员 (..\src\LDAWrap.cc) [D:\de
v\opensource\node-opencv\build\opencv.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\zrs\AppData\Roaming\npm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\zrs\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--module_name=opencv" "--module_path=./build/opencv/v4.0.0/Release/node-v46-win32-x64" "--msvs_version=2013"
gyp ERR! cwd D:\dev\opensource\node-opencv
gyp ERR! node -v v4.2.2
gyp ERR! node-gyp -v v3.2.0
gyp ERR! not ok

--- a/README.md
+++ b/README.md
@@ -16,16 +16,18 @@ cool, I'd love to hear about it!
 You'll need OpenCV 2.3.1 or newer installed before installing node-opencv.
 
 ## Specific for Windows
-1. Download and install opencv (Be sure to use a 2.4 version) @
+1. Download and install OpenCV (Be sure to use a 2.4 version) @
 http://opencv.org/downloads.html
 For these instructions we will assume OpenCV is put at C:\OpenCV, but you can
 adjust accordingly.
 
-2. Add the following to your PATH variable
-    C:\OpenCV\build\x64\vc12\bin;
-   The "x64" needs to match the version of NodeJS you are using.
+2. If you haven't already, create a system variable called OPENCV_DIR and set it
+   to C:\OpenCV\build\x64\vc12
 
-   Also set OPENCV_DIR to C:\OpenCV\build
+   Make sure the "x64" part matches the version of NodeJS you are using.
+
+   Also add the following to your system PATH
+        ;%OPENCV_DIR%\bin
 
 3. Install Visual Studio 2013. Make sure to get the C++ components.
    You can use a different edition, just make sure OpenCV supports it, and you

--- a/README.md
+++ b/README.md
@@ -1,4 +1,4 @@
-# node-opencv 
+# node-opencv
 
 [![Build Status](https://secure.travis-ci.org/peterbraden/node-opencv.png)](http://travis-ci.org/peterbraden/node-opencv)
 
@@ -16,41 +16,25 @@ cool, I'd love to hear about it!
 You'll need OpenCV 2.3.1 or newer installed before installing node-opencv.
 
 ## Specific for Windows
-1. Download Install opencv @ - (I used version 2.4.4)
+1. Download and install opencv (Be sure to use a 2.4 version) @
 http://opencv.org/downloads.html
-Put it in c:\opencv
+For these instructions we will assume OpenCV is put at C:\OpenCV, but you can
+adjust accordingly.
 
-2. Install python version 2.7 @
-http://www.python.org/download/releases/2.7/
-put it in c:\python27
+2. Add the following to your PATH variable
+    C:\OpenCV\build\x64\vc12\bin;
+   The "x64" needs to match the version of NodeJS you are using.
 
-3. install pkg-config by downloading the all in one bundle @ - (I used Gtk+ 3.6.4)
-http://www.gtk.org/download/win64.php
-put it in c:\pkg-config
+   Also set OPENCV_DIR to C:\OpenCV\build
 
-4. Add the following to your path variables
-C:\pkg-config\bin;C:\OpenCV\build\x64\vc11\bin;
+3. Install Visual Studio 2013. Make sure to get the C++ components.
+   You can use a different edition, just make sure OpenCV supports it, and you
+   set the "vcxx" part of the variables above to match.
 
-5. Install visual-studio in 4 steps
-
-  - install Visual C++ 2010 Express
-  
-  - install Windows SDK for windows 7 and .net framework 4
-  
-  - install Visual Studio 2010 Service Pack 1
-  
-  - install Visual C++ 2010 Service Pack 1 Compiler
-
-
-6. Download npeterbraden/node-opencv fork
+4. Download peterbraden/node-opencv fork
 git clone https://github.com/peterbraden/node-opencv
 
-7. edit file src/Matrix.cpp
-put "inline double round( double d ) { return floor( d + 0.5);}" below "cv::Rect* setRect(Local<Object> objRect, cv::Rect &result);"
-
-8. run npm install
-
-Then:
+5. run npm install
 
 ```bash
 $ npm install opencv
@@ -175,8 +159,8 @@ im.line([x1,y1], [x2, y2])
 
 #### Object Detection
 
-There is a shortcut method for 
-[Viola-Jones Haar Cascade](http://www.cognotics.com/opencv/servo_2007_series/part_2/sidebar.html) object 
+There is a shortcut method for
+[Viola-Jones Haar Cascade](http://www.cognotics.com/opencv/servo_2007_series/part_2/sidebar.html) object
 detection. This can be used for face detection etc.
 
 ```javascript
@@ -239,4 +223,3 @@ contours.convexHull(index, clockwise);
 ## MIT License
 The library is distributed under the MIT License - if for some reason that
 doesn't work for you please get in touch.
-

--- a/binding.gyp
+++ b/binding.gyp
@@ -24,12 +24,12 @@
       ],
 
       "libraries": [
-        "<!@(pkg-config --libs opencv)"
+        "<!@(node utils/find-opencv.js --libs)"
       ],
       # For windows
 
       "include_dirs": [
-        "<!@(pkg-config --cflags opencv)",
+        "<!@(node utils/find-opencv.js --cflags)",
         "<!(node -e \"require('nan')\")"
       ],
 
@@ -45,8 +45,10 @@
         }],
         [ "OS==\"win\"", {
             "cflags": [
-              "<!@(pkg-config --cflags \"opencv >= 2.4.9\" )",
               "-Wall"
+            ],
+            "defines": [
+                "WIN"
             ],
             "msvs_settings": {
               "VCCLCompilerTool": {

--- a/src/OpenCV.h
+++ b/src/OpenCV.h
@@ -1,6 +1,16 @@
 #ifndef __NODE_OPENCV_H__
 #define __NODE_OPENCV_H__
 
+#ifdef WIN
+    /*
+        This is needed on Windows for Visual Studio to not throw an error in the
+        build/include/opencv2/flann/any.h file in OpenCV.
+    */
+    namespace std{ typedef type_info type_info; }
+#endif
+
+
+
 #include <v8.h>
 #include <node.h>
 #include <node_object_wrap.h>

--- a/utils/find-opencv.js
+++ b/utils/find-opencv.js
@@ -1,0 +1,115 @@
+var exec = require("child_process").exec;
+var fs = require("fs");
+var flag = process.argv[2];
+
+function main(){
+
+    //Try using pkg-config, but if it fails and it is on Windows, try the fallback
+    exec("pkg-config opencv " + flag, function(error, stdout, stderr){
+        if(error){
+            if(process.platform === "win32"){
+                fallback();
+            }
+        }
+        else{
+            console.log(stdout);
+        }
+    });
+}
+
+//======================Windows Specific=======================================
+
+function fallback(){
+    exec("echo %OPENCV_DIR%", function(error, stdout, stderr){
+        stdout = cleanupEchoOutput(stdout);
+        if(error){
+            console.log("ERROR: There was an error reading OPENCV_DIR");
+        }
+        else if(stdout === "%OPENCV_DIR%") {
+            console.log("ERROR: OPENCV_DIR doesn't seem to be defined");
+        }
+        else {
+            getVisualStudioVersion(function(version){
+                getProcessArch(function(bits){
+                    printPaths(stdout, version, bits);
+                });
+            });
+        }
+    });
+}
+
+function printPaths(opencvPath, version, bits){
+    if(flag === "--cflags") {
+        console.log("\"" + opencvPath + "include\"");
+        console.log("\"" + opencvPath + "include\\opencv\"");
+    }
+    else if(flag === "--libs") {
+        var libPath = opencvPath + bits + "\\vc" + version + "\\lib\\";
+
+        fs.readdir(libPath, function(err, files){
+            if(err){
+                console.log("ERROR: couldn't read the lib directory");
+                console.log(err);
+            }
+
+            var libs = "";
+            for(var i = 0; i < files.length; i++){
+                if(getExtension(files[i]) === "lib"){
+                    libs = libs + " \"" + libPath + files[i] + "\" \r\n ";
+                }
+            }
+            console.log(libs);
+        });
+    }
+    else {
+        console.log("Error: unknown argument '" + flag + "'");
+    }
+}
+
+//This gets the architecture of the NodeJS that is running this script,
+//either x86 or x64
+function getProcessArch(cb){
+    exec("echo %PROCESSOR_ARCHITECTURE%", function(error, stdout, stderr) {
+        if(!error) {
+            var arch = cleanupEchoOutput(stdout);
+            if(arch === "AMD64"){
+                cb("x64");
+            }
+            else if(arch === "x86"){
+                cb("x86");
+            }
+            else {
+                console.log("ERROR: Unrecognized architecture");
+            }
+        }
+        else {
+            console.log("ERROR: There was an error getting the architecture");
+        }
+    });
+}
+
+function getVisualStudioVersion(cb, version){
+    if(typeof(version) === "undefined") version = 11;
+    exec("reg query HKEY_CLASSES_ROOT\\VisualStudio.DTE." + version + ".0",
+        function(error, stdout, stderr){
+            if(!error){
+                cb(version);
+            }
+            else if(version < 13) {
+                //Try the next version
+                getVisualStudioVersion(cb, version + 1);
+            }
+            else {
+                console.log("ERROR: Can't find Visual Studio");
+            }
+        });
+}
+
+function cleanupEchoOutput(s){
+    return s.slice(0, s.length - 2);
+}
+
+function getExtension(s){
+    return s.substr(s.lastIndexOf(".") + 1);
+}
+main();

--- a/utils/find-opencv.js
+++ b/utils/find-opencv.js
@@ -3,12 +3,14 @@ var fs = require("fs");
 var flag = process.argv[2];
 
 function main(){
-
     //Try using pkg-config, but if it fails and it is on Windows, try the fallback
     exec("pkg-config opencv " + flag, function(error, stdout, stderr){
         if(error){
             if(process.platform === "win32"){
                 fallback();
+            }
+            else{
+                throw new Error("ERROR: pkg-config couldn't find OpenCV");
             }
         }
         else{
@@ -23,33 +25,28 @@ function fallback(){
     exec("echo %OPENCV_DIR%", function(error, stdout, stderr){
         stdout = cleanupEchoOutput(stdout);
         if(error){
-            console.log("ERROR: There was an error reading OPENCV_DIR");
+            throw new Error("ERROR: There was an error reading OPENCV_DIR");
         }
         else if(stdout === "%OPENCV_DIR%") {
-            console.log("ERROR: OPENCV_DIR doesn't seem to be defined");
+            throw new Error("ERROR: OPENCV_DIR doesn't seem to be defined");
         }
         else {
-            getVisualStudioVersion(function(version){
-                getProcessArch(function(bits){
-                    printPaths(stdout, version, bits);
-                });
-            });
+            printPaths(stdout);
         }
     });
 }
 
-function printPaths(opencvPath, version, bits){
+function printPaths(opencvPath){
     if(flag === "--cflags") {
-        console.log("\"" + opencvPath + "include\"");
-        console.log("\"" + opencvPath + "include\\opencv\"");
+        console.log("\"" + opencvPath + "\\..\\..\\include\"");
+        console.log("\"" + opencvPath + "\\..\\..\\include\\opencv\"");
     }
     else if(flag === "--libs") {
-        var libPath = opencvPath + bits + "\\vc" + version + "\\lib\\";
+        var libPath = opencvPath + "\\lib\\";
 
         fs.readdir(libPath, function(err, files){
             if(err){
-                console.log("ERROR: couldn't read the lib directory");
-                console.log(err);
+                throw new Error("ERROR: couldn't read the lib directory " + err);
             }
 
             var libs = "";
@@ -62,47 +59,8 @@ function printPaths(opencvPath, version, bits){
         });
     }
     else {
-        console.log("Error: unknown argument '" + flag + "'");
+        throw new Error("Error: unknown argument '" + flag + "'");
     }
-}
-
-//This gets the architecture of the NodeJS that is running this script,
-//either x86 or x64
-function getProcessArch(cb){
-    exec("echo %PROCESSOR_ARCHITECTURE%", function(error, stdout, stderr) {
-        if(!error) {
-            var arch = cleanupEchoOutput(stdout);
-            if(arch === "AMD64"){
-                cb("x64");
-            }
-            else if(arch === "x86"){
-                cb("x86");
-            }
-            else {
-                console.log("ERROR: Unrecognized architecture");
-            }
-        }
-        else {
-            console.log("ERROR: There was an error getting the architecture");
-        }
-    });
-}
-
-function getVisualStudioVersion(cb, version){
-    if(typeof(version) === "undefined") version = 11;
-    exec("reg query HKEY_CLASSES_ROOT\\VisualStudio.DTE." + version + ".0",
-        function(error, stdout, stderr){
-            if(!error){
-                cb(version);
-            }
-            else if(version < 13) {
-                //Try the next version
-                getVisualStudioVersion(cb, version + 1);
-            }
-            else {
-                console.log("ERROR: Can't find Visual Studio");
-            }
-        });
 }
 
 function cleanupEchoOutput(s){

